### PR TITLE
Add FAIR data and RO-Crate module

### PR DESCRIPTION
**Pull request by stain**
Introduces RO-Crate as a step-by-step tutorial.

This will replace step #2 "Publishing my first crate" of https://gallantries.github.io/video-library/modules/ro-crate

## Licensing

I have provided `license` Apache 2.0, as this material originates from https://www.researchobject.org/ro-crate/1.2-DRAFT/introduction.html developed by RO-Crate community contributors. I added a License section at bottom as Apache License requires this text to be present. Is there a way to show the `copyright` on top as well?

@hexylena indicated that we may not need to change the License for training material, Apache-2.0 should also be open access. 

However if it is the case for adding to this portal then I would need to gather relicensing approval from @ptsefton @simleo @bedroesb and others, or rather self-host on `researchobject.org` -- but not sure how to pull out the nice CSS tricks.

### CODE REVIEW

**Comment by stain** [comment]
Will undo the `Gemfile` changes in this PR -- I had to do those as it failed to build locally otherwise.

**Comment by hexylena** [comment]
The license is indeed linked from the metadata header. If this sort of notice (linking out to the spdx copy) is insufficient then we can absolutely find a better solution.



it can be in a box at the top. does it need to be opened? or is it sufficient to have it collapsed, and on click, present the license text?

**Comment by hexylena** [comment]
The license is indeed linked from the metadata header. If this sort of notice (linking out to the spdx copy) is insufficient then we can absolutely find a better solution. We are indeed missing the "copyright" line which, yes, needs to be added.



it can be in a box at the top. does it need to be opened? or is it sufficient to have it collapsed, and on click, present the license text?

**Comment by hexylena** [comment]
> @hexylena indicated that we may not need to change the License for training material, Apache-2.0 should also be open access.
>
> However if it is the case for adding to this portal then I would need to gather relicensing approval from @ptsefton @simleo @bedroesb and others, or rather self-host on researchobject.org -- but not sure how to pull out the nice CSS tricks.

No, Apache-2.0 is **completely fine**, **no relicensing required**. We have existing MIT licensed materials, that was why we added support for annotating material-specific licenses in the first place.

**Comment by hexylena** [comment]
> Is there a way to show the copyright on top as well?

We can add this to the metadata, yes.

Additionally you've placed it in the wrong topic, I'll go ahead and move it.

File: topics/data-science/tutorials/ro-crate-intro/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,642 @@
...
+- RO-Crate provides a structure to make FAIR data packages
+- schema.org in JSON-LD provides a controlled vocabulary for FAIR metadata
+- Each entity of the crate is described separately
+- Cross-references between entities create a graph
+- The RO-Crate specification recommends which types and keys to use
+
+subtopic: fair
+contributors:
+  - stain
+  - ptsefton
```
Comment (hexylena) [inline topics/data-science/tutorials/ro-crate-intro/tutorial.md:47]: If you like can use our fancy new contributions syntax to more accurately record contributions to a training material, declaring which people should be recorded for authorship or just editing.

```
contributions:
  authorship:
    - shiltemann
  editing:
    - bebatut
    - hexylena
  funding:
    - carpentries
  testing:
    - userX
  ux:
    - userY
  infrastructure:
    - userZ
```

see the [full gtn tutorial schema documentation](https://training.galaxyproject.org/training-material/topics/contributing/tutorials/schemas/tutorial.html) for details on the individual roles.

**Comment by hexylena** [comment]
GTN changes required to pass tests are in https://github.com/ResearchObject/galaxy-training-material/pull/1

**Comment by stain** [comment]
@hexylena thanks for your help, I merged in https://github.com/ResearchObject/galaxy-training-material/pull/1 now.

I am not sure about the replacement PNG and Markdown for Figure 2, while the accessibility alt text is there, the fonts were converted wrongly (the arrow does not line up);  I can regenerate that if PNG really is preferred. The folder image also got a bit big/blurry.

The other thing is the text within the PNG is not selectable, as it was in the SVG. That is the reason I used the `<object>` tag (which also permitted the font loading but unfortunately also Javascript). Perhaps this is not a big thing as the JSON text is also there in the solutions of the previous step.

I see many others lessons like bash-git are using SVGs which are scalable. If there's reluctance to keep the `<object>` tag mainly, then I have converted now the text to paths, so it won't need the font loading, and kept it in Markdown syntax. 

It would still not be selectable/screen-readable in img, but should render consistently.  See related discussion in jgm/pandoc#7906

Attached screenshot of PNG vs SVG with some text selected while it was inserted as `<object>` :

screenshot

**Comment by hexylena** [comment]
> The other thing is the text within the PNG is not selectable, as it was in the SVG. That is the reason I used the <object> tag (which also permitted the font loading but unfortunately also Javascript). Perhaps this is not a big thing as the JSON text is also there in the solutions of the previous step.
>
> I see many others lessons like bash-git are using SVGs which are scalable. If there's reluctance to keep the <object> tag mainly, then I have converted now the text to paths, so it won't need the font loading, and kept it in Markdown syntax.

No this all makes sense. This was very much a "let's get this shipped in an 'acceptable' format soon" compromise to get it out of the door, rather than an enforced "this is how it must be".

SVGs are fine in general, and of course preferable where possible due to file sizes. Thus, we clearly should add some support for the object tag.  Having selectable text is definitely nice.

For me it was rendering quite small (though I block a lot of external resources, and was perhaps also blocking the styling that you applied to the SVG).

> It would still not be selectable/screen-readable in img, but should render consistently. See related discussion in https://github.com/jgm/pandoc/issues/7906

Let's ship the consistent rendering version now, and today I'll work on supporting <objects> so we can do something more proper by tomorrow.

**Comment by hexylena** [comment]
looks like a few invalid links. We do not have a good way to make exceptions to the link checking policy (intentionally), could we manually edit this index.html to fix/remove the invalid ones?

```
* At ./_site/training-material/topics/fair/tutorials/ro-crate-intro/rainfall-1.2.1/index.html:220:

  internally linking to #ro-crate-metadata.json; the file exists, but the hash 'ro-crate-metadata.json' does not

* At ./_site/training-material/topics/fair/tutorials/ro-crate-intro/rainfall-1.2.1/index.html:283:

  internally linking to #https%3A//creativecommons.org/licenses/by-nc-sa/4.0/; the file exists, but the hash 'https%3A//creativecommons.org/licenses/by-nc-sa/4.0/' does not

* At ./_site/training-material/topics/fair/tutorials/ro-crate-intro/rainfall-1.2.1/ro-crate-preview.html:220:

  internally linking to #ro-crate-metadata.json; the file exists, but the hash 'ro-crate-metadata.json' does not

* At ./_site/training-material/topics/fair/tutorials/ro-crate-intro/rainfall-1.2.1/ro-crate-preview.html:283:

  internally linking to #https%3A//creativecommons.org/licenses/by-nc-sa/4.0/; the file exists, but the hash 'https%3A//creativecommons.org/licenses/by-nc-sa/4.0/' does not
```

**Comment by hexylena** [comment]
Ahh a javascript dependency, yes, that can be it. Our link checker doesn't execute JS.

**Review by hexylena — APPROVED** [review]

**Pull request merged**